core/control-panel/impl/src/models/artifact.rs

```diff
@@ -0,0 +1,107 @@
+use crate::core::ic_cdk::next_time;
+use orbit_essentials::{
+    storable,
+    types::{Timestamp, UUID},
+    utils::sha256_hash,
+};
+use uuid::Uuid;
+
+/// The artifact id, which is a UUID.
+pub type ArtifactId = UUID;
+
+/// The artifact is a record that is stored in the artifact repository.
+#[storable]
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Artifact {
+    /// The UUID identifies an artifact.
+    id: ArtifactId,
+    /// The artifact byte size.
+    size: u64,
```

Comment:
Why do we store it instead of calling `artifact.len()`?

Reply:
It was mostly added for clarify, but since we need to load the actual content to be able to access that maybe we could just add that to the artifact api instead of storing that as a value

Reply:
> It was mostly added for clarify, but since we need to load the actual content to be able to access that maybe we could just add that to the artifact api instead of storing that as a value

Indeed. Unlike hashing, this should be constant time to evaluate if I'm not mistaken.